### DISCUSSION TITLE
Use './dsreplication enable ' to add  OpenDj-4.3.3 server, but  can't query the status of replication

### DISCUSSION THREAD

**Discussion by yoland-f** [OpenIdentityPlatform/OpenDJ#451] [Q&A]
### Description:

**1. There are four servers in Opendj  replication 。I add anther server into them.**
   [root@yh bin]# ./dsreplication enable --adminUID admin --adminPassword password --baseDN dc=gtja,dc=net --host1 10.172.16.147  --port1 4444 --bindDN1 "cn=DirectT" --bindPassword1 PWD --replicationPort1 8989 --host2 10.172.16.237 --port2 4444 --bindDN2 "cn=DirectT" --bindPassword2 PWD --replicationPort2 8989 --trustAll --no-prompt

Establishing connections ..... Done.
Checking registration information ..... Done.
Updating remote references on server 10.172.16.147:4444 ..... Done.
Configuring Replication port on server 10.172.16.237:4444 ..... Done.
Updating replication configuration for baseDN dc=gtja,dc=net on server
10.172.16.147:4444 .....Done.
Updating replication configuration for baseDN dc=gtja,dc=net on server
10.172.16.150:4444 .....Done.
Updating remote references on server 10.172.16.150:4444 ..... Done.
Updating replication configuration for baseDN dc=gtja,dc=net on server
10.172.16.148:4444 .....Done.
Updating remote references on server 10.172.16.148:4444 ..... Done.
Updating replication configuration for baseDN dc=gtja,dc=net on server
10.172.16.149:4444 .....Done.
Updating remote references on server 10.172.16.149:4444 ..... Done.
Updating replication configuration for baseDN dc=gtja,dc=net on server
10.172.16.237:4444 .....Done.
Updating registration configuration on server 10.172.16.147:4444 ..... Done.
Updating registration configuration on server 10.172.16.150:4444 ..... Done.
Updating registration configuration on server 10.172.16.148:4444 ..... Done.
Updating registration configuration on server 10.172.16.149:4444 ..... Done.
Updating registration configuration on server 10.172.16.237:4444 ..... Done.
Updating replication configuration for baseDN cn=schema on server
10.172.16.147:4444 .....Done.
Updating replication configuration for baseDN cn=schema on server
10.172.16.150:4444 .....Done.
Updating replication configuration for baseDN cn=schema on server
10.172.16.148:4444 .....Done.
Updating replication configuration for baseDN cn=schema on server
10.172.16.149:4444 .....Done.
Updating replication configuration for baseDN cn=schema on server
10.172.16.237:4444 .....Done.
Initializing registration information on server 10.172.16.237:4444 with the
contents of server 10.172.16.147:4444 .....Done.
Initializing schema on server 10.172.16.237:4444 with the contents of server
10.172.16.147:4444 .....Done.

Replication has been successfully enabled.  Note that for replication to work
you must initialize the contents of the base DNs that are being replicated
(use dsreplication initialize to do so).
See /tmp/opendj-replication-6821654084514691793.log for a detailed log of this
operation.

**2.When  first add the cluster, I can check the status.** 
[root@yh bin]# ./dsreplication status  --port 4444   --hostname 10.172.16.237 --adminUID admin --adminPassword password --trustAll --no-prompt
Suffix DN      : Server              : Entries : Replication enabled : DS ID : RS ID : RS Port (1) : M.C. (2) : A.O.M.C. (3) : Security (4)
---------------:---------------------:---------:---------------------:-------:-------:-------------:----------:--------------:-------------
dc=gtja,dc=net : 10.172.16.147:4444 : 11254   : true                : 19292 : 25162 : 8989        : 0        :              : false
dc=gtja,dc=net : 10.172.16.148:4444 : 11254   : true                : 3781  : 5377  : 8989        : 0        :              : false
dc=gtja,dc=net : 10.172.16.149:4444 : 11254   : true                : 11523 : 17232 : 8989        : 0        :              : false
dc=gtja,dc=net : 10.172.16.150:4444 : 11254   : true                : 23630 : 6904  : 8989        : 0        :              : false
dc=gtja,dc=net : 10.172.16.237:4444 : 27      : true                : 13122 : 29976 : 8989        : 0        :              : false

[1] The port used to communicate between the servers whose contents are being
replicated.
[2] The number of changes that are still missing on this server (and that have
been applied to at least one of the other servers).
[3] Age of oldest missing change: the date on which the oldest change that has
not arrived on this server was generated.
[4] Whether the replication communication through the replication port is
encrypted or not.

**3.But after 2 minutes, the status query returned a failure. And the content of the log is empty.**
[root@yh bin]# ./dsreplication status  --port 4444   --hostname 10.172.16.237 --adminUID admin --adminPassword password --trustAll --no-prompt
Error reading registration information.  Details: The provided value
"6442450929" could not be parsed as an integer
See /tmp/opendj-replication-3617825614643382274.log for a detailed log of this
operation.

### 
Question：
**Why can't query the status?And how to slove the problem?**
Thank you for your answer first.

**Comment by vharseko** [comment]
try 
```
./dsreplication initialize --baseDN dc=gtja,dc=net --adminUID admin --adminPassword password --hostSource 10.172.16.147 --portSource 4444 --hostDestination 10.172.16.237 --portDestination 4444 -X -n
```

**Reply by yoland-f** [reply]
**I try the command,But it still  fail.I have three questions.**
Hope for answer.

**1. Why is this error "Error reading registration information. Details: The provided value "6442450929" could not be parsed as an integer"  occured?Is there any data in the OpenDj server uncorrect?**

**2. Is there any log switch ?Because there is no log in the file.**

**3.When query the status on 237,there is an error.Why can't query the status?**
[root@yh bin]# ./status  -D "cn=DirectT" -w PWD 
Exception in thread "main" java.lang.NullPointerException
	at org.opends.guitools.controlpanel.datamodel.ControlPanelInfo.regenerateDescriptor(ControlPanelInfo.java:522)
	at org.opends.server.tools.status.StatusCli.execute(StatusCli.java:310)
	at org.opends.server.tools.status.StatusCli.mainCLI(StatusCli.java:176)
	at org.opends.server.tools.status.StatusCli.main(StatusCli.java:139)

**Comment by vharseko** [comment]
> Why is this error "Error reading registration information. Details: The provided value "6442450929" could not be parsed as an integer" occured?Is there any data in the OpenDj server uncorrect

value "6442450929" in database or config out of range

> Is there any log switch ?Because there is no log in the file.
check 
* /tmp/opendj-replication-6821654084514691793.log
* /opt/opendj/log 
or use [Enabling Debug Logging](https://doc.openidentityplatform.org/opendj/admin-guide/chap-troubleshooting#troubleshoot-enable-debug-logging)

> When query the status on 237,there is an error.Why can't query the status?

please /opt/opendj/bin/stop-ds  10.172.16.237 and attach /opt/opendj full zip folder for reproduce

**Reply by yoland-f** [reply]
> value "6442450929" in database or config out of range

How can I find which  record or which  config?

**Reply by vharseko** [reply]
for restore instance: drop and recreate replica

for self diagnostic: 
* [Enabling Debug Logging](https://doc.openidentityplatform.org/opendj/admin-guide/chap-troubleshooting#troubleshoot-enable-debug-logging)
* [Debug instance](https://github.com/OpenIdentityPlatform/OpenDJ/wiki/How-To#debug-instance)

for community diagnostic: stop-ds instance and attach /opt/opendj full zip folder
